mongodb/vibe/db/mongo/connection.d

```diff
@@ -181,6 +183,56 @@ final class MongoConnection {
 			throw new MongoDriverException(format("Failed to connect to MongoDB server at %s:%s.", m_settings.hosts[0].name, m_settings.hosts[0].port), __FILE__, __LINE__, e);
 		}
 
+		if (m_settings.nextgen) {
+			import os = std.system;
+			import compiler = std.compiler;
+			string platform = compiler.name ~ " "
+				~ compiler.version_major.to!string ~ "." ~ compiler.version_minor.to!string;
+			// TODO: add support for os.version
+
+			version (X86_64)
+				string arch = "x86_64 ";
+			else version (X86)
+				string arch = "x86 ";
+			else version (AArch64)
+				string arch = "aarch64 ";
+			else version (ARM_HardFloat)
+				string arch = "armhf ";
+			else version (ARM)
+				string arch = "arm ";
+			else version (PPC64)
+				string arch = "ppc64 ";
+			else version (PPC)
+				string arch = "ppc ";
+			else
+				string arch = "unknown ";
```

Comment:
Better move this into a dedicated function. It might also be useful for other parts (e.g. request user-agent)